hw/xbox/mcpx_apu.c

```diff
@@ -826,12 +845,38 @@ static void se_frame(void *opaque)
                     NV_PAVS_VOICE_PAR_STATE_ACTIVE_VOICE)) {
                 MCPX_DPRINTF("voice %d not active...!\n", d->regs[current]);
                 fe_method(d, SE2FE_IDLE_VOICE, d->regs[current]);
+            } else {
+                process_voice(d, mixbin, d->regs[current]);
             }
             MCPX_DPRINTF("next voice %d\n", d->regs[next]);
             d->regs[current] = d->regs[next];
         }
     }
 
+#if INJECT_VP_BEEP
+    /* Inject some audio to the mixbin for debugging.
+     * Signal is 1500 Hz sine wave, phase shifted by mixbin number.*/
+    for (unsigned int i = 0; i < NUM_MIXBINS; i++) {
+        for (unsigned int j = 0; j < NUM_SAMPLES_PER_FRAME; j++) {
```

Comment:
I chose `int mixbin` and `int sample`; I'm not really happy about this change, because I feel like non-C99 loops are error prone, and I'll probably have to add some linebreaks to future PRs.

Also we might want to do `uint32_t* mixbin = &mixbins[i][0];` which is now.. bad.. because it mixes indices and pointers to subsets of an array.

I also had to rename the original `mixbin` variable to `mixbins` (which is a good change though).